R/opq.R

```diff
@@ -76,17 +88,30 @@
 #'     add_osm_feature (key = "place")
 #' opqa2 <- osmdata_sf (qa2)
 #' }
-opq <- function (bbox = NULL, nodes_only = FALSE,
+opq <- function (bbox = NULL,  nodes_only = FALSE,
+                 out = c ("body", "tags", "meta", "skel", "tags center", "ids"),
                  datetime = NULL, datetime2 = NULL,
                  timeout = 25, memsize) {
 
     timeout <- format (timeout, scientific = FALSE)
     prefix <- paste0 ("[out:xml][timeout:", timeout, "]")
+
     suffix <- ifelse (
         nodes_only,
         "); out;",
         ");\n(._;>;);\nout body;"
     ) # recurse down
+    out <- try (match.arg (out))
+    if (inherits (out, "try-error")) {
+        stop ('out parameter must be "body", "tags", "meta", "skel", "tags center" or "ids".')
+    }
+    has_geometry <- out %in% c ("body", "meta", "skel")
+    if (has_geometry && out != "body") {
+        suffix <- gsub ("out( body)*;", paste0 ("out ", out, ";"), suffix)
+    } else if (!has_geometry) {
+        suffix <- gsub ("\\);(\\n\\(\\._;>;\\);\\n| )out( body)*;", paste0 ("); out ", out, ";"), suffix)
```

Comment:
Why not just redefine `suffix` completely?
```suggestion
        suffix <- paste0 ("); out ", out, ";")
```